### PR TITLE
Instructor Help Pages: replace magic numbers used as element IDs #8240

### LINKED ISSUE #8240: Instructor Help Pages: replace magic numbers used as element IDs
For example, `src/main/webapp/partials/instructorHelpFAQ.jsp` has the code `<li id="faq2">`. The number `2` here is a _magic number_.  
Perhaps we can replace them with more meaningful words e.g. `<li id="faq-change-details-after-enrolling">` etc.?

### PR DESCRIPTION
**Pull request by dabbler011**
Fixes #8240 

Replaced magic numbers used as element ids and updated their links accordingly.

### CODE REVIEW

**Comment by dabbler011** [comment]
@damithc Can you please have a look at this?

File: src/main/webapp/instructorHelp.jsp [hunk 1]
```diff
@@ -14,47 +14,47 @@
...
-              <a href="#faq11">How are the contribution numbers calculated?</a>
+              <a href="#faq-how-contribution-numbers-calculated">How are the contribution numbers calculated?</a>
             </li>
             <li>
-              <a href="#faq12">Is it compulsory for students to use Google accounts?</a>
+              <a href="#faq-is-google-account-usage-compulsory-for-student">Is it compulsory for students to use Google accounts?</a>
             </li>
             <li>
-              <a href="#faq13">What if my course doesn’t have teams?</a>
+              <a href="#faq-if-course-have-no-teams">What if my course doesn’t have teams?</a>
```
Comment (damithc) [inline src/main/webapp/instructorHelp.jsp]: have --> has

File: src/main/webapp/instructorHelp.jsp [hunk 2]
```diff
@@ -14,47 +14,47 @@
         <div style="margin: 0 auto; padding: 0 50px;">
           <ul>
             <li>
-              <a href="#faq1">How do I view a list of students in a course?</a>
+              <a href="#faq-students-in-a-course">How do I view a list of students in a course?</a>
```
Comment (damithc) [inline src/main/webapp/instructorHelp.jsp]: `view-student-list-in-course` ?

File: src/main/webapp/instructorHelp.jsp [hunk 3]
```diff
@@ -14,47 +14,47 @@
...
-              <a href="#faq5">How do I change the Google ID associated with a student?</a>
+              <a href="#faq-change-googleID-of-student">How do I change the Google ID associated with a student?</a>
             </li>
             <li>
-              <a href="#faq6">What to do if a student says his/her courses have disappeared from the system?</a>
+              <a href="#faq-if-courses-displayed-to-students-disappear">What to do if a student says his/her courses have disappeared from the system?</a>
             </li>
             <li>
-              <a href="#faq7">How do I interpret/use contribution values in the results of ‘Team contribution’ type questions?</a>
+              <a href="#faq-use-contribution-values-in-result-of-Team-comtribution">How do I interpret/use contribution values in the results of ‘Team contribution’ type questions?</a>
```
Comment (damithc) [inline src/main/webapp/instructorHelp.jsp]: `interpret-contribution-values-in-results`?

File: src/main/webapp/instructorHelp.jsp [hunk 4]
```diff
@@ -14,47 +14,47 @@
...
             </li>
             <li>
-              <a href="#faq8">What to do if a student says he cannot submit responses due to some technical glitch?</a>
+              <a href="#faq-if-students-cannot-submit-an-evaluation">What to do if a student says he cannot submit responses due to some technical glitch?</a>
             </li>
             <li>
-              <a href="#faq9">How come the actual contribution values entered by the student is different from the values shown in the results?</a>
+              <a href="#faq-actual-contribution-by-student-and-values-in-result-differ">How come the actual contribution values entered by the student is different from the values shown in the results?</a>
```
Comment (damithc) [inline src/main/webapp/instructorHelp.jsp]: `entered-contribution-differs-from-results`?

File: src/main/webapp/instructorHelp.jsp [hunk 5]
```diff
@@ -14,47 +14,47 @@
...
-              <a href="#faq10">Can a student influence his ‘perceived contribution’ value by awarding himself a larger contribution?</a>
+              <a href="#faq-can-student-influence-his-contribution">Can a student influence his ‘perceived contribution’ value by awarding himself a larger contribution?</a>
             </li>
             <li>
-              <a href="#faq11">How are the contribution numbers calculated?</a>
+              <a href="#faq-how-contribution-numbers-calculated">How are the contribution numbers calculated?</a>
             </li>
             <li>
-              <a href="#faq12">Is it compulsory for students to use Google accounts?</a>
+              <a href="#faq-is-google-account-usage-compulsory-for-student">Is it compulsory for students to use Google accounts?</a>
```
Comment (damithc) [inline src/main/webapp/instructorHelp.jsp]: `is-google-account-compulsory-for-student`

**Comment by damithc** [comment]
Also remember to update the branch

**Comment by xpdavid** [comment]
@dabbler011 Have you checked these places since you have changed the IDs?

image

You can do a global search in the page in case you forget something.

**Comment by dabbler011** [comment]
@damithc @xpdavid done. see if it is all right now.

File: src/main/webapp/partials/instructorHelpSessions.jsp [hunk 6]
```diff
@@ -1018,7 +1018,7 @@
 
       <div class="bs-example">
         <form class="form-horizontal form_question" role="form">
-          <div class="panel panel-primary questionTable" id="questionTable1">
+          <div class="panel panel-primary questionTable" id="intrHelpQuestionTable">
```
Comment (xpdavid) [inline src/main/webapp/partials/instructorHelpSessions.jsp]: You cannot use same `id` in the same page. Though it works, but it is not HTML5 compliant.

File: src/main/webapp/partials/instructorHelpSessions.jsp [hunk 7]
```diff
@@ -2094,7 +2094,7 @@
           <br>
           <div class="bs-example">
             <form class="form-horizontal form_question" role="form">
-              <div class="panel panel-primary questionTable" id="questionTable1">
+              <div class="panel panel-primary questionTable" id="intrHelpQuestionTable">
```
Comment (xpdavid) [inline src/main/webapp/partials/instructorHelpSessions.jsp]: Same and there are more...

**Comment by dabbler011** [comment]
@xpdavid removed duplicacies and also deployed at https://teammates-akshat.appspot.com/

**Comment by wkurniawan07** [comment]
> Looks like our id _(add: and class)_ naming style is not consistent

@damithc that has been a problem for months/years now...

**Pull request merged**